staging/src/k8s.io/kubectl/pkg/util/deployment/deployment.go

```diff
@@ -181,10 +181,9 @@ func (o replicaSetsByCreationTimestamp) Less(i, j int) bool {
 
 // // FindOldReplicaSets returns the old replica sets targeted by the given Deployment, with the given slice of RSes.
 // // Note that the first set of old replica sets doesn't include the ones with no pods, and the second set of old replica sets include all old replica sets.
-func findOldReplicaSets(deployment *appsv1.Deployment, rsList []*appsv1.ReplicaSet) ([]*appsv1.ReplicaSet, []*appsv1.ReplicaSet) {
+func findOldReplicaSets(deployment *appsv1.Deployment, rsList []*appsv1.ReplicaSet, newRS *appsv1.ReplicaSet) ([]*appsv1.ReplicaSet, []*appsv1.ReplicaSet) {
```

Comment:
I don't think we need to update this interface. It is unnecessary.

Reply:
findOldReplicaSets is an un-exported func.
newRS is needed by this func.

Can you clarify how newRS is passed to this func without changing the func ?

Reply:
I mean just keeping current two methods as they are. They are both local methods. I think it's okay to keep them unchanged.

I am open for the changes if you prefer the newer interface.

Reply:
The newer interface calculates newRS once.
I prefer the newer interface.